src/datachain/lib/signal_schema.py

```diff
@@ -464,19 +464,63 @@ def contains_file(self) -> bool:
         return False
 
     def slice(
-        self, keys: Sequence[str], setup: Optional[dict[str, Callable]] = None
+        self,
+        params: dict[str, Union[DataType, Any]],
+        setup: Optional[dict[str, Callable]] = None,
+        is_batch: bool = False,
     ) -> "SignalSchema":
```

Comment:
Main changes of this PR goes into this function.

`params` is now a dictionary with param names as keys and types as values instead of just a list of input params names.

`is_batch` is needed to be able to check batching UDF input params types.

Let's say we do have a signal `foo` with type `int`, then regular UDF might looks like this:

```python
def process(foo: int) -> int:
    ...
```

and batching UDF looks like this:

```python
def process_batch(foo: list[int]) -> int:
    ...
```

To be able to check types here we need to know if UDF is batching or not.